apps/heft/src/cli/HeftActionRunner.ts

```diff
@@ -494,7 +569,15 @@ export class HeftActionRunner {
         changedFiles,
         globChangedFilesFn,
         fileEventListener
-      ).then(() => false);
+      ).then(
+        () => false,
+        (error: unknown) => {
+          // Re-throw the error so that it doesn't get swallowed. This way we can ensure that the list of
+          // changed files is not cleared, since we will want to resurface the error on future incremental
+          // builds.
+          throw error;
+        }
```

Comment:
This should technically do the same thing as not passing this parameter at all. Curious what behavior you are seeing?